src/Files.App/Views/Shells/ModernShellPage.xaml.cs

```diff
@@ -52,6 +53,13 @@ public ModernShellPage() : base(new CurrentInstanceViewModel())
 			_navigationInteractionTracker.NavigationRequested += OverscrollNavigationRequested;
 		}
 
+		private void ShellViewModel_DirectoryInfoUpdated(object sender, EventArgs e)
+		{
+			// Regular binding causes issues when refreshing the directory so we set the text manually
+			if (FilterTextBox is not null && FilterTextBox.IsLoaded)
```

Comment:
Only if you like this:
```suggestion
			if (FilterTextBox?.IsLoaded ?? false)
```